pointpats/centrography.py

```diff
@@ -590,95 +590,122 @@ def _(points: GeoPandasBase) -> np.float64:
     return std_distance(coords)
 
 
+
 @singledispatch
-def ellipse(points):
+def ellipse(points, weights=None, method="crimestat", 
+              crimestatCorr=True, degfreedCorr=True):
+    """
+    Computes a weighted standard deviational ellipse for a set of point geometries.
+
+    Parameters:
+        points : arraylike
+                 array representing a point pattern
+        weights : arraylike
+                 array of weights for each point (Optional)
+        method (str): Correction method to apply. Must be 'crimestat' or 'yuill'.
+        crimestatCorr (bool): Apply CrimeStat correction if method == 'yuill'.
+        degfreedCorr (bool): Apply degrees-of-freedom correction if method == 'yuill'.
+
+    Returns:
+        tuple: (major_axis_length, minor_axis_length, rotation_angle)
```

Comment:
Can you make sure this follows numpydoc formatting? This will not render properly in docs.